ui/pages/routes/confirmation-handler.tsx

```diff
@@ -94,13 +120,16 @@ export const ConfirmationHandler = () => {
     swapsFetchParams,
   ]);
 
+  const isExemptedRoute = EXEMPTED_ROUTES.some((route) =>
+    pathname.startsWith(route),
+  );
+
   const hasAllowedPopupRedirectApprovals = pendingApprovals.some((approval) =>
     SNAP_APPROVAL_TYPES.includes(approval.type),
   );
 
   useEffect(() => {
-    // Only run when on home/default page (for now)
-    if (pathname !== DEFAULT_ROUTE) {
+    if (isExemptedRoute) {
```

Comment:
Popup
- No change since it always start in the home screen

Fullscreen
- Same filters as previous

Sidepanel
- Will listen for approvals except while on exempted routes